### PR TITLE
TIP-713: Fix remover

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**

Fixes behats:
- `features/update/add_multiselect.feature:6`
- `features/update/add_price.feature:6`
- `features/update/remove_price.feature:6`
- `features/update/set_price.feature:6` => I changed the results of some behats because I find them better now than before imo. (no business rule broken)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Ok
| Added Behats                      | Ok
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Component/Catalog/Updater/Remover/PriceCollectionAttributeRemover.php [hunk 1]
```diff
@@ -85,15 +85,20 @@ protected function removePrices(ProductInterface $product, AttributeInterface $a
...
 
+        $currencyToRemove = [];
+        foreach ($data as $priceToRemove) {
+            $currencyToRemove[] = $priceToRemove['currency'];
+        }
+
         if (null !== $productValue) {
             $prices = [];
             foreach ($productValue->getData() as $price) {
-                if (!in_array(['amount' => $price->getData(), 'currency' => $price->getCurrency()], $data)) {
```
Comment (SamirBoulil) [inline src/Pim/Component/Catalog/Updater/Remover/PriceCollectionAttributeRemover.php:91]: 👀  In my fix, I only check for the currency to remove the price value. not the price *and* the value.

What do you think ?

File: features/update/remove_price.feature [hunk 2]
```diff
@@ -16,7 +16,7 @@ Feature: Remove price fields
       | BOX4 | 5 EUR, 5 USD  | 5 EUR, 5 USD  |
     Then I should get the following products after apply the following updater to it:
       | product | actions                                                                                                                                                                                                                                                | result                                                                                                                                                                                                                          |
-      | BOX1    | [{"type": "remove_data", "field": "price", "data": [{"amount": "", "currency": "EUR"}], "locale": "fr_FR", "scope": null}]                                                                                                                             | {"values": {"price": [{"locale": "fr_FR", "scope": null, "data": [{"amount": "5", "currency": "USD"}]}, {"locale": "en_US", "scope": null, "data": [{"amount": "5", "currency": "EUR"}, {"amount": "5", "currency": "USD"}]}]}} |
```
Comment (SamirBoulil) [inline features/update/remove_price.feature:19]: The before and after results are the same. I just inverted the order in wich the prices in the price collection appears.

File: features/update/set_price.feature [hunk 3]
```diff
@@ -16,10 +16,10 @@ Feature: Update price fields
...
       | BOX4    | [{"type": "set_data", "field": "price", "data": [{"amount": "5", "currency": "USD"}], "locale": "fr_FR", "scope": null}]                                                                                                                         | {"values": {"price": [{"locale": "fr_FR", "scope": null, "data": [{"amount": 5, "currency": "USD"}]}]}}                                      |
-      | BOX4    | [{"type": "set_data", "field": "price", "data": [{"amount": "5", "currency": "EUR"}], "locale": "fr_FR", "scope": null}, {"type": "set_data", "field": "price", "data": [{"amount": "5", "currency": "USD"}], "locale": "fr_FR", "scope": null}] | {"values": {"price": [{"locale": "fr_FR", "scope": null, "data": [{"amount": null, "currency": "EUR"}, {"amount": 5, "currency": "USD"}]}]}} |
```
Comment (SamirBoulil) [inline features/update/set_price.feature:24]: Here, the new price collection generated from the updated does not keep the prices set to null. and generates a clean price collection with actual values in it.

File: src/Pim/Component/Catalog/Updater/Remover/PriceCollectionAttributeRemover.php (hunk 1 above)
Comment (damien-carcel) [reply src/Pim/Component/Catalog/Updater/Remover/PriceCollectionAttributeRemover.php:91]: That is a PO question. If the collection contains 4 € and you ask to remove 5 €, what should happen?

File: src/Pim/Component/Catalog/Updater/Remover/PriceCollectionAttributeRemover.php [hunk 4]
```diff
@@ -85,15 +85,20 @@ protected function removePrices(ProductInterface $product, AttributeInterface $a
...
+                if (!in_array($price->getCurrency(), $currencyToRemove)) {
                     $prices[] = ['amount' => $price->getData(), 'currency' => $price->getCurrency()];
                 }
             }
 
-            $this->productBuilder->addOrReplaceProductValue($product, $attribute, $scope, $locale, $prices);
+            $this->productBuilder->addOrReplaceProductValue($product, $attribute, $locale, $scope, $prices);
```
Comment (damien-carcel) [inline src/Pim/Component/Catalog/Updater/Remover/PriceCollectionAttributeRemover.php:101]: :+1:

**Review by damien-carcel — APPROVED** [review]
GTM, but before merging, you should ask a PO regarding your concern about checking only currencies, and if it can have an impact on API.

File: src/Pim/Component/Catalog/Updater/Remover/PriceCollectionAttributeRemover.php (hunk 1 above)
Comment (SamirBoulil) [reply src/Pim/Component/Catalog/Updater/Remover/PriceCollectionAttributeRemover.php:91]: In fact, this is how 1.7 is working. It does not check the amount but removes the price for the given currency: https://github.com/akeneo/pim-community-dev/blob/ba189044dd93fb14042de97ba6a0a2ce8382e642/src/Pim/Component/Catalog/Updater/Remover/PriceCollectionAttributeRemover.php#L84-L84

Comment (damien-carcel) [reply src/Pim/Component/Catalog/Updater/Remover/PriceCollectionAttributeRemover.php:91]: Then OK in this case.

**Review by jjanvier — APPROVED** [review]

**Review by grena — APPROVED** [review]

**Pull request merged**